census_view.cpp

```diff
@@ -1540,9 +1541,73 @@ table_type_converter const& table_type_converter::get_impl()
 
 } // Unnamed namespace.
 
+/// The attribute provider to make all odd rows appear grey.
+
+class CensusViewGridCellAttrProvider
+    :public wxGridCellAttrProvider
+{
+  public:
+    CensusViewGridCellAttrProvider()
+    {
+        lightGray_ = wxColor(0xf7f7f7);
+        attrForEvenRows_ = new wxGridCellAttr();
+        attrForOddRows_ = new wxGridCellAttr();
+        attrForOddRows_->SetBackgroundColour(lightGray_);
+    }
+
+    ~CensusViewGridCellAttrProvider() override
+    {
+        attrForOddRows_->DecRef();
```

Comment:
Please use `wxObjectDataPtr<wxGridCellAttr>` instead of manual `DecRef()` calls. They're error prone and I'm not sure if there is no memory leak here even now (why isn't `attrForEvenRows_` decref'd?).